lib/improver/cli/nbhood_land_and_sea.py

```diff
@@ -32,118 +32,21 @@
-                       'should be used. For example: 10000,12000,14000 1,2,3 '
-                       'where a lead time of 1 hour uses a radius of 10000m, '
-                       'a lead time of 2 hours uses a radius of 12000m, etc.')
-    parser.add_argument('--sum_or_fraction', default="fraction",
-                        choices=["sum", "fraction"],
-                        help='The neighbourhood output can either be in the '
-                        'form of a sum of the neighbourhood, or a '
-                        'fraction calculated by dividing the sum of the '
-                        'neighbourhood by the neighbourhood area. '
-                        '"fraction" is the default option.')
-    parser.add_argument('--intermediate_filepath', default=None,
-                        help='Intermediate filepath for results following '
-                        'topographic masked neighbourhood processing of '
-                        'land points and prior to collapsing the '
-                        'topographic_zone coordinate. Intermediate files '
-                        'will not be produced if no topographic masked '
-                        'neighbourhood processing occurs.')
-
-    args = parser.parse_args(args=argv)
-
-    cube = load_cube(args.input_filepath)
-    mask = load_cube(args.input_mask_filepath, no_lazy_load=True)
-    weights = None
-    if any(['topographic_zone' in coord.name()
-            for coord in mask.coords(dim_coords=True)]):
-
-        if mask.attributes['topographic_zones_include_seapoints'] == 'True':
-            raise ValueError('The topographic zones mask cube must have been '
-                             'masked to exclude sea points, but '
-                             'topographic_zones_include_seapoints = True')
-
-        if not args.weights_for_collapsing_dim:
-            raise IOError('A weights cube must be provided if using a mask '
-                          'of topographic zones to collapse the resulting '
-                          'vertical dimension.')
-
-        weights = load_cube(args.weights_for_collapsing_dim, no_lazy_load=True)
-
-    result, intermediate_cube = process(
-        cube, mask, args.radius, args.radii_by_lead_time,
-        weights, args.sum_or_fraction, args.intermediate_filepath)
-
-    save_netcdf(result, args.output_filepath)
-    if args.intermediate_filepath:
-        save_netcdf(intermediate_cube, args.intermediate_filepath)
-
-
-def process(cube, mask, radius=None, radii_by_lead_time=None, weights=None,
-            sum_or_fraction="fraction", return_intermediate=False):
+from improver import cli
+
+
+# pylint: disable=too-many-branches, too-many-locals
+@cli.clizefy
+@cli.with_output
+@cli.with_intermediate_output
+def process(cube: cli.inputcube,
+            mask: cli.inputcube,
+            weights: cli.inputcube = None,
+            *,
+            radius: float = None,
+            radii_by_lead_time=None,
```

Comment:
This argument isn't tested at all in the acceptance tests.

I also know this setup isn't right because I'm trying to set up a "list of values" argument for standardise in the same way, and it's failing!

We should discuss...

Reply:
Not going to comment on what is/isn't tested, but do have some views here on how the CLI should look.

Short term, we need two options: `radius`, and `lead_times` - if `lead_times` is provided, then `radius` needs to be a list of the same length etc..
Longer term, we need to determine if `improver` can work with single cubes containing multiple lead times - if not, and we truly are stuck with the "one time cube per file" thing, then we should just accept a single option, `radius`, because we can pass a varying radius in (with each lead time) from elsewhere (e.g: the suite).

Reply:
(@cgsandford on "a list of values", does the `comma_separated_values" thing do as you want? See the `percentiles-to-realizations` CLI for an example use case)

Reply:
Ah, quite possibly.  Will test on Monday.

Reply:
Comment here remains valid: @neilCrosswaite could you correct this to be the appropriate variable type (possibly `comma_separated_list`, but I think it's more complicated than that) and add a test for it please.

Reply:
We have never tested radii by the looks of it, so i might need some help setting that up.

Reply:
(Agreed with @neilCrosswaite that writing extra tests can be follow-on work)

Reply:
I am a bit concerned by leaving this in an untested state, as it is going to make the suite migration potentially more difficult.  Although the CLI tests apparently only cover `radius`, neighbourhooding (although I'm not quite sure which neighbourhooding) in the suite uses the `radii-with-lead-time` option in several places.

I think we need to check whether this option is used from this CLI within the suite, and if so, we should fix this as an immediate follow-on PR.  @fionaRust or @bayliffe may be able to help with this @neilCrosswaite , otherwise I have a little time tomorrow morning if you're desperate.

Reply:
This test has now been made.